lib/SampleService/core/config.py

```diff
@@ -207,15 +231,46 @@ def get_validators(url: str) -> MetadataValidatorSet:
     :returns: A set of metadata validators.
     '''
     # TODO VALIDATOR make validator CLI
+
     try:
-        with _urllib.request.urlopen(url) as res:
-            cfg = _yaml.safe_load(res)
+        config_asset = None
+        if url:
+            config_url = url
+        elif not repo_path:
+            raise ValueError(f'No metadata validator config URL or repo path.')
+        else:
+            try: 
+                repo = _Github(login_or_token=token).get_repo(repo_path)
+                releases = [rel for rel in repo.get_releases() if prerelease_ok or not rel.prerelease]
+            except:
+                raise RuntimeError(f'Fetching releases from repo {repo_path} failed.')
+            if not releases:
+                raise ValueError(f'No releases found in validator config repo {repo_path}')
+            latest_release = releases[0] # max(releases, key=lambda rel: rel.created_at)
+            assets = latest_release.get_assets()
+            if not assets:
+                raise ValueError(f'No assets found in validator config repo {repo_path}')
+            config_asset = next((a for a in assets if a.name==repo_file), None)
+            if not config_asset:
+                raise ValueError(f'No config asset found in validator config repo {repo_path}')
+            config_url = config_asset.url
+
+        req = request.Request(config_url)
+        req.add_header('Accept', 'application/octet-stream')
+        if token:
+            req.add_header('Authorization', f'token {token}')
+        with request.urlopen(req) as response:
+            cfg = _yaml.safe_load(response)
+
     except _URLError as e:
-        raise ValueError(
-            f'Failed to open validator configuration file at {url}: {str(e.reason)}') from e
+        if config_asset:
+            raise ValueError(f'Error downloading config asset from {config_asset.url}: {str(e.reason)}') from e
+        else:
+            raise ValueError(f'Error downloading config asset from {url or repo_path}: {str(e.reason)}') from e
```

Comment:
If repo_path, include the other relevant info - release & filename info

Reply:
Here just the repo_path was removed - I don't think that was the intent of the change?

Reply:
never mind, see comment below

Reply:
:+1:

Reply:
it was yes, see my reply below